matrix_sdk_crypto/src/machine.rs

```diff
@@ -731,14 +724,40 @@ impl OlmMachine {
     }
 
     /// Get a verification request object with the given flow id.
-    pub fn get_verification_request(&self, flow_id: &EventId) -> Option<VerificationRequest> {
+    pub fn get_verification_request(
+        &self,
+        flow_id: impl AsRef<str>,
+    ) -> Option<VerificationRequest> {
         self.verification_machine.get_request(flow_id)
     }
 
     async fn update_one_time_key_count(&self, key_count: &BTreeMap<DeviceKeyAlgorithm, UInt>) {
         self.account.update_uploaded_key_count(key_count).await;
     }
 
+    async fn handle_to_device_evnet(&self, event: &AnyToDeviceEvent) {
+        match event {
+            AnyToDeviceEvent::RoomKeyRequest(e) => {
+                self.key_request_machine.receive_incoming_key_request(&e)
+            }
+            AnyToDeviceEvent::KeyVerificationAccept(..)
+            | AnyToDeviceEvent::KeyVerificationCancel(..)
+            | AnyToDeviceEvent::KeyVerificationKey(..)
+            | AnyToDeviceEvent::KeyVerificationMac(..)
+            | AnyToDeviceEvent::KeyVerificationRequest(..)
+            | AnyToDeviceEvent::KeyVerificationReady(..)
+            | AnyToDeviceEvent::KeyVerificationDone(..)
+            | AnyToDeviceEvent::KeyVerificationStart(..) => {
+                self.handle_verification_event(&event).await;
+            }
+            AnyToDeviceEvent::Dummy(_) => {}
+            AnyToDeviceEvent::RoomKey(_) => {}
+            AnyToDeviceEvent::ForwardedRoomKey(_) => {}
+            AnyToDeviceEvent::RoomEncrypted(_) => {}
+            AnyToDeviceEvent::Custom(_) => {}
```

Comment:
Aren't these supposed to be handled eventually? Maybe a TODO comment?

Reply:
`Dummy` doesn't need to be handled, `RoomKey` and `ForwardedRoomKey` are handled separately since we zero out private key data and only allow them to be received as a previous `RoomEncrypted` event. `RoomEncrypted` is handled separately so we avoid encryption bombs, i.e. an encrypted event inside an encrypted event inside an...